### PR TITLE
Logging: Adds x-cache response header when present

### PR DESCRIPTION
**Pull request by owensmallwood**
fixes https://github.com/grafana/grafana-enterprise/issues/4616

**Changes**
Adds the `X-Cache` response header to the context log when present.

Reasons for change described on the issue.

**How to test**
- make sure query caching is enabled
- cause some panel queries that get served by the cache
- look at your context logs and you should see either an `x-cache=HIT` or `x-cache=MISS` label

### CODE REVIEW

File: devenv/docker/blocks/loki-promtail/promtail-config.yaml [hunk 1]
```diff
@@ -22,4 +22,4 @@ scrape_configs:
       - localhost
     labels:
       job: grafana
-      __path__: /var/log/grafana/*log
\ No newline at end of file
+      __path__: /var/log/grafana/*log
```
Comment (gabor) [inline devenv/docker/blocks/loki-promtail/promtail-config.yaml:25]: sorry, is this change intentional?

Comment (owensmallwood) [reply devenv/docker/blocks/loki-promtail/promtail-config.yaml:25]: no, this was GoLand. Not sure why it did this

Comment (owensmallwood) [reply devenv/docker/blocks/loki-promtail/promtail-config.yaml:25]: should probably have a newline at the end of the file anyways

**Review by hairyhenderson — COMMENTED** [review]
LGTM, but I think a feature flag should be introduced for this.
